docs/rtk-query/usage/prefetching.mdx

```diff
@@ -89,10 +89,10 @@ export function usePrefetchImmediately<T extends EndpointNames>(
   arg: Parameters<typeof api.endpoints[T]['initiate']>[0],
   options: PrefetchOptions = {}
 ) {
-  const dispatch = useDispatch()
+  const dispatch = useAppDispatch()
   useEffect(() => {
-    dispatch(api.util.prefetch(endpoint, arg, options))
-  }, [])
+    dispatch(api.util.prefetch(endpoint, arg as any, options))
+  }, [arg, dispatch, endpoint, options])
```

Comment:
If we add a dependency array here, we will need to keep `arg` and `options` stable. 
Either we add a note to the docs or we export the tools we have in the `query/react` package.

@markerikson what are your thoughts on exposing `useShallowStableValue` and maybe even `useStableQueryArgs` (the latter would need changes to be usable outside of the hooks internals)?

Reply:
Yea, I had that in the description, but wasn't sure if we need to talk about keeping those things stable as a part of this work or not. 100% agree with exposing `useShallowStableValue` :)

Reply:
@markerikson Any thoughts here? I can always just revert this back to a dep array of `[]` just so we can get the `arg as any` in.

Reply:
too many other things going on, no thoughts here :)

Reply:
keep it with the `[]` for now and we'll think of the rest for the next release when we have resources?